app/models/task.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  id          :uuid             not null, primary key
+#  name        :string
+#  description :string
+#  state       :integer          default(0)
+#  project_id  :uuid
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_tasks_on_project_id  (project_id)
+#
+
+class Task < ActiveRecord::Base
+  belongs_to :project
+
+  enum state: [:todo, :in_progress, :done]
+
+  validates :name, presence: true, uniqueness: true
+  validates :state, presence: true
+  validates :project_id, presence: true
```

Comment:
You don't do any validating of the project in the controller or a service, so the better thing to do here would be to validate the presence of `project` instead of `project_id`. That way Rails will verify the project exists as part of validation. When you just validate the presence of the id you can stick anything in there whether the project actually exists or not.
Whether you have the validation of project existence here in the model or at a higher level like the controller (or a service) it doesn't matter too much and is up to your style, but it belongs somewhere.